modules/extension/xsd/xsd-core/src/main/java/org/geotools/xsd/impl/BindingPropertyExtractor.java

```diff
@@ -116,28 +115,28 @@ public List properties(Object object, XSDElementDeclaration element) {
                 // turn each map entry into a particle
                 HashMap particles = new HashMap();
 
-                for (Iterator e = map.entrySet().iterator(); e.hasNext(); ) {
-                    Map.Entry entry = (Map.Entry) e.next();
+                for (Iterator e = map.keySet().iterator(); e.hasNext(); ) {
```

Comment:
why change to a keyset if you are going to do a map.get(key) later anyways?

Reply:
The new Map type doesn't have an `entrySet()` method, and this was the least ugly way to do it. `mapIterator` for example has different semantics. I'm not too sure on the details anymore, but I remember spending some time pulling my hair out on this change.

Reply:
As an alternative, the MultiValuedMap can return a MapIterator, which returns the key when calling "next", and has a dedicated "getValue" method to fetch the value associated to the current entry.